Wire Auth0 token interceptor and file edit/delete flow in CRBM-Viz

gmarupilla opened the pull request:
I added client id and domain to config and kept it in gitignore. I've added config.example.json which the user can copy paste as config.json and modify the contents for testing purposes. Or environment wise.

Commit:
Merge branch 'master' into development

Commit:
Get token on login

Commit:
Merge branch 'master' into development

Commit:
Change auth0 credentials

Commit:
Remove auth service from about component

Commit:
Get auth0 credentials from config file

Commit:
Merge remote-tracking branch 'upstream/master' into development

Commit:
Remove merge conflict marker

Commit:
Fix imports in app.module

Commit:
Fix merge markers in routing module

Commit:
Fix merge markers in file table component

Commit:
Add example config

Commit:
Merge branch 'master' into development

Commit:
Get Id token instead of access token

bilalshaikh42 commented:
The build and docker tests are failing. Could you make sure that the app builds properly outside your local environment? 
@gmarupilla

gmarupilla commented:
I've added `config.example.json` as a default config.. Or a template config..every time we want to build it.. We will copy example config to `config.json`
And then it'll work. It's a replacement for environment variable. Since in front-end app we cannot access environment variables

Commit:
Add file upload and file view to links

bilalshaikh42 commented:
> I've added `config.example.json` as a default config.. Or a template config..every time we want to build it.. We will copy example config to `config.json`
> And then it'll work. It's a replacement for environment variable. Since in front-end app we cannot access environment variables

I see that the config file contains just the Auth0 client id. In that case, it does not need to be hidden and should be included in the source to allow for automatic building and deployment

https://community.auth0.com/t/client-id-vs-secret/9558

Commit:
Edit functionality working

Commit:
Delete functionality working

Commit:
Create file service, update file list on upload/delete file

Commit:
Remove unused imports

Commit:
Prevent refresh on load of FileEditComponent, redirect to file list on save

Commit:
Create simulation service to handle simulation data from backend

Commit:
Move file edit requests to file service, remove reference of hardcoded config, remove config ts file

Commit:
Get sbatch files in component

Commit:
Add loading spinner to past simulations

Commit:
Add loading spinner to new simulation

Commit:
Add loading spinner to file table

Commit:
Change location of config file

Commit:
Modify dockerfile as per new config structure

gmarupilla commented:
> > I've added `config.example.json` as a default config.. Or a template config..every time we want to build it.. We will copy example config to `config.json`
> > And then it'll work. It's a replacement for environment variable. Since in front-end app we cannot access environment variables
> 
> I see that the config file contains just the Auth0 client id. In that case, it does not need to be hidden and should be included in the source to allow for automatic building and deployment
> 
> https://community.auth0.com/t/client-id-vs-secret/9558

Still, I think it's better to have such things in a configuration file rather than being hardcoded into code itself. Here's one use case why, I have my own personal auth0 account which I use to experiment/test the code that I write, if I modify client id in code, I'll need to exempt it from commits every single time I change anything else. With a configuration file like this, user can modify configuration with ease according to different environments. To resolve this, I added docker rule to copy config.example.json (which is default config for heroku) to config.json. Moreover, more things are also added to config, and will keep on increasing in the future.

bilalshaikh42 commented:
> > > I've added `config.example.json` as a default config.. Or a template config..every time we want to build it.. We will copy example config to `config.json`
> > > And then it'll work. It's a replacement for environment variable. Since in front-end app we cannot access environment variables
> > 
> > 
> > I see that the config file contains just the Auth0 client id. In that case, it does not need to be hidden and should be included in the source to allow for automatic building and deployment
> > https://community.auth0.com/t/client-id-vs-secret/9558
> 
> Still, I think it's better to have such things in a configuration file rather than being hardcoded into code itself. Here's one use case why, I have my own personal auth0 account which I use to experiment/test the code that I write, if I modify client id in code, I'll need to exempt it from commits every single time I change anything else. With a configuration file like this, user can modify configuration with ease according to different environments. To resolve this, I added docker rule to copy config.example.json (which is default config for heroku) to config.json. Moreover, more things are also added to config, and will keep on increasing in the future.

Got it. In that case, lets make use of the environment.ts and environment.prod.ts files in the environments folder. The auth service has a good example on how I used the files to change based on various environments. You could even add a new environment for your development environment. Thi way, everything needed to build the app (currently the build step is failing in the node workflow) is present in default configurations at all times. When building, use the --configuration flag
Here is a good guide for it:
[Link]( https://blog.angularindepth.com/becoming-an-angular-environmentalist-45a48f7c20d8)

Commit:
Replace config with environment

Commit:
Remove copy config rule from Dockerfile

Commit:
Merge branch 'master' into development

gmarupilla commented:
> > > > I've added `config.example.json` as a default config.. Or a template config..every time we want to build it.. We will copy example config to `config.json`
> > > > And then it'll work. It's a replacement for environment variable. Since in front-end app we cannot access environment variables
> > > 
> > > 
> > > I see that the config file contains just the Auth0 client id. In that case, it does not need to be hidden and should be included in the source to allow for automatic building and deployment
> > > https://community.auth0.com/t/client-id-vs-secret/9558
> > 
> > 
> > Still, I think it's better to have such things in a configuration file rather than being hardcoded into code itself. Here's one use case why, I have my own personal auth0 account which I use to experiment/test the code that I write, if I modify client id in code, I'll need to exempt it from commits every single time I change anything else. With a configuration file like this, user can modify configuration with ease according to different environments. To resolve this, I added docker rule to copy config.example.json (which is default config for heroku) to config.json. Moreover, more things are also added to config, and will keep on increasing in the future.
> 
> Got it. In that case, lets make use of the environment.ts and environment.prod.ts files in the environments folder. The auth service has a good example on how I used the files to change based on various environments. You could even add a new environment for your development environment. Thi way, everything needed to build the app (currently the build step is failing in the node workflow) is present in default configurations at all times. When building, use the --configuration flag
> Here is a good guide for it:
> [Link](https://blog.angularindepth.com/becoming-an-angular-environmentalist-45a48f7c20d8)

Thanks for the info, moved configs to environment files.

gmarupilla commented:
I guess tests are failing in the components you've written. Can you look into it?

bilalshaikh42 commented:
Yup tests are not yet passing per #9 .  For now we can only assume it works if building. I'll merge this in

bilalshaikh42 commented on CRBM-Viz/angular.json:
```diff
@@ -63,6 +63,35 @@
+                  "with": "src/environments/environment.dev.ts"
+                }
+              ],
+              "optimization": true,
```
This might be very slow for development. Up to you, but we can change the optimization and AOT off since we dont be deploying using this environment

bilalshaikh42 commented on CRBM-Viz/src/app/Components/loading-spinner/loading-spinner.component.html:
```diff
@@ -0,0 +1,13 @@
+<div class="container">
+    <div class="row">
+        <div class="span12" style="height: 80vh; padding-top: 30vh; padding-left: 30vw;">
```
Can you move the styles to the sass? If there is a reason for it to be here, its fine in that case

bilalshaikh42 commented on CRBM-Viz/src/app/Components/loading-spinner/loading-spinner.component.sass:
```diff
@@ -0,0 +1,49 @@
+@keyframes lds-eclipse
```
Angular material already has a spinner defined. 
Can we use that instead for simplicity and consistancy? 
https://material.angular.io/components/progress-spinner/overviewhttps://material.angular.io/components/progress-spinner/overview

bilalshaikh42 commented on CRBM-Viz/src/app/Layout/links/links.component.ts:
```diff
@@ -8,5 +9,13 @@ import { AuthService } from 'src/app/Services/auth0.service';
 export class LinksComponent implements OnInit {
   constructor(public auth: AuthService) {}
 
-  ngOnInit() {}
```
As mentioned in #29 this component will be deleted, so it is not a good idea to have the profile load here. In any case, it is probably not a good idea to have the profile load in an unrelated component. Can this be moved to the main app component?

bilalshaikh42 commented on CRBM-Viz/src/app/Services/auth0.service.ts:
```diff
@@ -12,6 +12,7 @@ import {
 import { tap, catchError, concatMap, shareReplay } from 'rxjs/operators';
 import { Router } from '@angular/router';
 import { environment } from 'src/environments/environment';
+import { environment as config }  from 'src/environments/environment';
```
This is just messy. Lets change the config name below to environment

bilalshaikh42 commented on CRBM-Viz/src/app/Services/auth0.service.ts:
```diff
@@ -22,8 +23,8 @@ export class AuthService {
     createAuth0Client({
-      domain: 'crbm.auth0.com',
-      client_id: '0NKMjbZuexkCgfWY3BG9C3808YsdLUrb',
+      domain: config.auth0.domain,
```
Change this to environment.autho0

bilalshaikh42 commented on CRBM-Viz/src/app/Services/auth0.service.ts:
```diff
@@ -121,9 +140,10 @@ export class AuthService {
       // Call method to log out
       client.logout({
-        client_id: '0NKMjbZuexkCgfWY3BG9C3808YsdLUrb',
+        client_id: config.auth0.clientId,
```
Likewise here

Commit:
Renamed config to environment

Commit:
Add material spinner

Commit:
Move auth token subscription from links component to app component

Commit:
Merge branch 'master' into development

Commit:
Import unimported statements

bilalshaikh42 committed:
Change default login

bilalshaikh42 committed:
Fixed file service tests

bilalshaikh42 committed:
Clean ip app module imports, fix simulate component

bilalshaikh42 committed:
fixed FIel table component tests

bilalshaikh42 committed:
Check for token before setting local storage

bilalshaikh42 committed:
add todo
